src/services/codefixes/fixUnusedIdentifier.ts

```diff
@@ -11,33 +11,37 @@ namespace ts.codefix {
         Diagnostics.All_destructured_elements_are_unused.code,
         Diagnostics.All_variables_are_unused.code,
     ];
+
     registerCodeFix({
         errorCodes,
         getCodeActions(context) {
             const { errorCode, sourceFile, program } = context;
             const checker = program.getTypeChecker();
-            const startToken = getTokenAtPosition(sourceFile, context.span.start, /*includeJsDocComment*/ false);
+            const sourceFiles = program.getSourceFiles();
+            const token = getTokenAtPosition(sourceFile, context.span.start, /*includeJsDocComment*/ true);
 
-            const importDecl = tryGetFullImport(startToken);
+            const importDecl = tryGetFullImport(token);
             if (importDecl) {
                 const changes = textChanges.ChangeTracker.with(context, t => t.deleteNode(sourceFile, importDecl));
                 return [createCodeFixAction(fixName, changes, [Diagnostics.Remove_import_from_0, showModuleSpecifier(importDecl)], fixIdDelete, Diagnostics.Delete_all_unused_declarations)];
             }
-            const delDestructure = textChanges.ChangeTracker.with(context, t => tryDeleteFullDestructure(t, sourceFile, startToken, /*deleted*/ undefined, checker, /*isFixAll*/ false));
+            const delDestructure = Deleter.with(context, d =>
+                tryDeleteFullDestructure(token, d, checker, sourceFiles, /*isFixAll*/ false));
             if (delDestructure.length) {
                 return [createCodeFixAction(fixName, delDestructure, Diagnostics.Remove_destructuring, fixIdDelete, Diagnostics.Delete_all_unused_declarations)];
             }
-            const delVar = textChanges.ChangeTracker.with(context, t => tryDeleteFullVariableStatement(t, sourceFile, startToken, /*deleted*/ undefined));
+            const delVar = Deleter.with(context, d => tryDeleteFullVariableStatement(sourceFile, token, d));

```

Comment:
can we do this in the change tracker instead? the change tracker will track the changes, and will only do them at the end, so we can do the filtering of deleted ranges then instead of having a special deleter object.